File: src/main/java/pl/training/patterns/behavioral/interpreter/Variable.java
package pl.training.patterns.behavioral.interpreter;

import java.util.Map;

public class Variable implements Expression {
    private final String name;

    @Override
    public double evaluate(Map<String, Double> context) {
        return context.get(name);
    }

    public Variable(final String name) {
        this.name = name;
    }
}

File: csharp/src/Training.Patterns/Behavioral/Interpreter/Variable.cs
namespace Training.Patterns.Behavioral.Interpreter;

public class Variable : IExpression
{
    private readonly string name;

    public double Evaluate(IReadOnlyDictionary<string, double> context) => context[name];

    public Variable(string name)
    {
        this.name = name;
    }
}
